akka-http-core/src/main/resources/reference.conf

```diff
@@ -431,6 +443,18 @@ akka.http {
 
       # Enable verbose debug logging for all ingoing and outgoing frames
       log-frames = false
+
+      # When there is no data transmitted, send a HTTP/2 ping frame with this interval to make sure the connection is
+      # kept alive, 0s disables sending ping frames. Must be full seconds.
+      keepalive-time = 0s
+
+      # Fail the connection if a ping frame is sent and does not come back from the server within this timeout.
+      # Must be full seconds and less than the ping interval.
+      keepalive-timeout = 5s
+
+      # Fail the connection if more ping interchanges than this are done without any data or header frames.
+      # Setting to 0 disables this limit. Only in effect if ping-interval is enabled.
+      max-keepalives-without-data = 0
```

Comment:
What's the goal of this setting? A second level of guards, that still closes the connection after a while?

Reply:
We could also treat that as a different feature. Would probably be nicer to configure just with a timeout.

Reply:
Yeah, it corresponds with the gRPC GRPC_ARG_HTTP2_MAX_PINGS_WITHOUT_DATA setting, so I guess it would somewhat replaces/duplicates the idle connection timeout.